dynamorm/model.py

```diff
@@ -578,7 +578,9 @@ def update(self, conditions=None, update_item_kwargs=None, return_all=False, **k
             resp = self.update_item(conditions=conditions, update_item_kwargs=update_item_kwargs, **kwargs)
 
             # update our local attrs to match what we updated
-            for key, val in six.iteritems(resp['Attributes']):
+            partial_model = self.new_from_raw(resp['Attributes'], partial=True)
```

Comment:
Just for my own education, why do we need to do both a `new_from_raw` and the individual `setattr` calls?

Reply:
If we're guaranteed to get back all the attributes for the update call, we could probably just instantiate a new model from it. The change is just creating a partial object and patching the original with validated native types.